tzupdate.py

```diff
@@ -102,6 +102,41 @@ def link_localtime(timezone, zoneinfo_path, localtime_path):
     return zoneinfo_tz_path
 
 
+def export_etc_timezone(timezone, etc_timezone_path):
+    '''
+    Check whether the system uses a /etc/timezone file (or an equivalent
+    provided by the user).  If so, write the name of the timezone to that file.
+
+    Debian-based distributions (since Etc, i.e. 4.0) use a /etc/timezone file.
+    See:
+
+        https://wiki.debian.org/TimeZoneChanges
+        https://help.ubuntu.com/community/UbuntuTime
+
+    Return a boolean indicating whether the indicated file exists and was
+    overwritten.
+    '''
+    wrote_etc_timezone = False
+    if os.path.exists(etc_timezone_path):
+        wrote_etc_timezone = True
+        try:
+            with open(etc_timezone_path, 'w') as fd:
+                fd.write(timezone+'\n')
+        except (IOError,OSError) as thrown_exc:
+            # If we don't have permission to write /etc/timezone, we probably
+            # need to be root.
+            if thrown_exc.errno == errno.EACCES:
+                msg = 'Could not link "%s" (%s). Are you root?' % (
+                        etc_timezone_path, thrown_exc,
+                    )
+                thrown_exc.message = msg
+                raise thrown_exc
+            elif thrown_exc.errno != errno.ENOENT:
```

Comment:
[blocker] This is going to silently ignore any other error since you don't reraise under other conditions. Just remove this other elif.